### PR TITLE
initial files for rtl87x2g hal

### PR DESCRIPTION
**Pull request by ZhiyuanTang17**
This commit add initial files for rtl87x2g series HAL.
With this commit, building and running hello_world is supported.
For driver part, I have ported clock-control, pinmux, gpio and uart.
HI @YuzhuoLiuRTK , @RSGraceGuan, Please help review.

### CODE REVIEW

**Review by RSGraceGuan — APPROVED** [review]

File: bee/CMakeLists.txt [hunk 1]
```diff
@@ -1,2 +1,3 @@
-add_subdirectory_ifdef(drivers)
+add_subdirectory(drivers)
+zephyr_include_directories(drivers)
```
Comment (RSGraceGuan) [inline bee/CMakeLists.txt:2]: Is this line necessary?

File: bee/drivers/CMakeLists.txt [hunk 2]
```diff
@@ -1 +1,4 @@
+add_subdirectory_ifdef(CONFIG_RTK_UART uart)
 add_subdirectory_ifdef(CONFIG_RTK_GPIO gpio)
```
Comment (RSGraceGuan) [inline bee/drivers/CMakeLists.txt]: Please replace with CONFIG_REALTEK_UART.

File: bee/drivers/rcc/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,2 @@
+add_subdirectory(src)
```
Comment (RSGraceGuan) [inline bee/drivers/rcc/CMakeLists.txt:1]: Maybe select the corresponding source file directory according to the different IC is a better option.

File: bee/CMakeLists.txt (hunk 1 above)
Comment (ZhiyuanTang17) [reply bee/CMakeLists.txt:2]: This include directory is to solve this build error：
image

File: bee/drivers/rcc/CMakeLists.txt (hunk 3 above)
Comment (ZhiyuanTang17) [reply bee/drivers/rcc/CMakeLists.txt:1]: I designed to distinguish IC in **_CMakeList.txt in the src directory._**

**Comment by ZhiyuanTang17** [comment]
This PR may discard and @YuzhuoLiuRTK  will create a PR containing these files in this PR.
cc @RSGraceGuan

**Pull request closed without merging**